### DISCUSSION TITLE
datasize

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#955] [Q&A]
datasize.Size(node.MetaData.Size).MachineString() call fileinfo.Size() Why does the unit always return b instead of b, kb, gb based on size?

![Screenshot_20240502_102157_com.realvnc.viewer.android.jpg](https://github.com/cogentcore/core/assets/19886504/328cea32-708e-4b0c-a6bf-dfd3a877135d)

**Comment by kkoreilly** [comment]
Try using the `String` function instead of the `MachineString` function for user-facing data size display.

**Comment by ddkwork** [comment]
thx， working



---Original---
From: ***@***.***&gt;
Date: Thu, May 2, 2024 12:09 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [cogentcore/core] datasize (Discussion #955)




 
Try using the String function instead of the MachineString function for user-facing data size display.
